packages/core/src/html/UlNodeProcessor.ts

```diff
@@ -0,0 +1,189 @@
+import cheerio from 'cheerio';
+import { MbNode, NodeOrText } from '../utils/node';
+
+const { processIconString } = require('../lib/markdown-it/plugins/markdown-it-icons');
+const emojiDictionary = require('../lib/markdown-it/patches/markdown-it-emoji-fixed');
+
+interface EmojiData {
+  [key: string]: string;
+}
+
+const emojiData = emojiDictionary as unknown as EmojiData;
+
+const ICON_ATTRIBUTES = ['icon', 'i-width', 'i-height', 'i-size', 'i-class'];
+
+interface IconAttributes {
+  icon?: string;
+  className?: string;
+  size?: string;
+  width?: string;
+  height?: string;
+}
+
+type IconAttributeDetail = {
+  isFirst: boolean;
+  iconAttr: IconAttributes | null;
+  level: number;
+};
+
+type IconAttributeWrapper = {
+  key: number;
+  value: IconAttributeDetail;
+};
+
+function classifyIcon(icon: string) {
+  const isEmoji = Object.prototype.hasOwnProperty.call(emojiData, icon);
+
+  return {
+    isEmoji,
+    unicodeEmoji: isEmoji
+      ? emojiData[icon]
+      : undefined,
+  };
+}
+
+function createIconSpan(iconAttrs: IconAttributes): cheerio.Cheerio {
+  const {
+    isEmoji,
+    unicodeEmoji,
+  } = classifyIcon(iconAttrs.icon!);
+
+  let spanContent;
+  if (isEmoji) {
+    spanContent = `<span aria-hidden="true">${unicodeEmoji}</span>`;
+  } else {
+    spanContent = processIconString(iconAttrs.icon);
+  }
+
+  let spanNode;
+  if (spanContent === null) {
+    const img = cheerio(`<img src='${iconAttrs.icon}' alt='Icon'>`)
+      .css({ width: iconAttrs.width, height: iconAttrs.height, display: 'inline-block' })
+      .addClass(iconAttrs.className || '');
+    img.append('\u200B');
+
+    spanContent = cheerio('<span></span>').append(img).css({
+      'padding-bottom': '0.3em',
+      'padding-top': '0.3em',
+    });
+    spanNode = cheerio(spanContent).css({ 'font-size': 'unset' });
+  } else {
+    spanNode = cheerio(spanContent).css({ 'font-size': 'unset' });
+    spanNode = spanNode.css({ 'font-size': iconAttrs.size }).addClass(iconAttrs.className || '');
+  }
+  // Add invisible character to avoid the element from being empty
+  spanNode.append('\u200B');
+  return spanNode.css({
+    'line-height': 'unset',
+    'margin-inline-end': '0.3em',
+    'flex-shrink': '0',
```

Comment:
what's `flex-shrink` for?

Reply:
This is to prevent the icon width from shrinking if the content occupies too much space under the flexbox layout.
https://www.w3schools.com/CSS/tryit.asp?filename=trycss3_flexbox_item

Reply:
Do you have an example? I'm not able to reproduce this with `flex-shrink` deleted.

Reply:
Without `flex-shrink`
<img width="1884" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/655da9c9-2f41-410f-be89-cc84cf442547">

With `flex-shrink`
<img width="1734" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/ce85dc71-1ccb-48f9-9d6e-8ccf5ac5e44a">

Reply:
![image](https://github.com/MarkBind/markbind/assets/3306138/3ec33644-e657-4a8e-a2d6-447f26efeab8)

![image](https://github.com/MarkBind/markbind/assets/3306138/69642fe0-7701-4386-b220-12fe87c0a628)


This won't ever occur in the implementation, won't it?

Icons always use `font-size` and images have width/height on the inner element.

These are the 2 cases I tried:

![image](https://github.com/MarkBind/markbind/assets/3306138/862e5ac2-277c-449f-ab37-23228e5f1bb1)

![image](https://github.com/MarkBind/markbind/assets/3306138/84818359-4b96-49e9-a35b-9780dc27fda6)

Reply:
Yes, the flex-shrink was applied in the previous implementation where height and width were required. Since current dim purely relies on font-size, and the image is not a direct child of li, flex-shrink is now redundant.

Reply:
flex-shrink is need to maintain the dimension for image.
<img width="799" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/7f2e986f-d677-4429-a450-49d5c2639445">

Reply:
> flex-shrink is still needed to maintain the dimension for image.

TIL, seems due to the readdition of min-width from above thread: https://ishadeed.com/article/min-max-css/#:~:text=Zero%20With%20Flexbox-,The%20default%20value%20for,a%20flex%20item%20is%20equal%20to%20the%20size%20of%20its%20contents.,-According%20to%20the